x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/privilege_monitoring/data_sources/sync/integrations/update_detection/privileged_status_match.ts

```diff
@@ -0,0 +1,147 @@
+     */
+    if (!source.matchers?.length) {
+      dataClient.log(
+        'info',
+        `No matchers for source id=${source.id ?? '(unknown)'}. Returning 0 privileged users`
+      );
+      return [];
+    }
+
+    const esClient = dataClient.deps.clusterClient.asCurrentUser;
+    const script = buildMatcherScript(source.matchers[0]);
+
+    let afterKey: AfterKey | undefined;
+    let fetchMore = true;
+    const pageSize = 100; // number of agg buckets per page
+    const users: PrivMonIntegrationsUser[] = [];
+
+    try {
+      while (fetchMore) {
+        const response = await esClient.search<never, PrivMatchersAggregation>({
+          index: source.indexPattern, // Time range relies on sync markers: https://github.com/elastic/security-team/issues/13985
+          ...buildPrivilegedSearchBody(script, 'now-10y', afterKey, pageSize),
+        });
+
+        const aggregations = response.aggregations;
+        const privUserAgg = response.aggregations?.privileged_user_status_since_last_run;
+        const buckets = privUserAgg?.buckets ?? [];
+
+        // process current page
+        if (buckets.length && aggregations) {
+          const privMonUsers = await extractPrivMonUsers(aggregations, source.matchers[0]);
+          users.push(...privMonUsers);
+        }
+
+        // next page
+        afterKey = privUserAgg?.after_key;
+        fetchMore = Boolean(afterKey);
+      }
+
+      dataClient.log('info', `Found ${users.length} privileged users from matchers.`);
+      return users;
+    } catch (error) {
+      dataClient.log('error', `Error finding privileged users from matchers: ${error.message}`);
+      return [];
+    }
+  };
+
+  const extractPrivMonUsers = async (
+    aggregation: PrivMatchersAggregation,
+    matchers: Matcher
+  ): Promise<PrivMonIntegrationsUser[]> => {
+    const buckets: PrivBucket[] | undefined =
+      aggregation.privileged_user_status_since_last_run?.buckets;
+    if (!buckets) {
+      return [];
+    }
+
+    const batchUsernames = buckets.map((bucket) => bucket.key.username);
+    const existingUserMap = await searchService.getExistingUsersMap(uniq(batchUsernames));
+
+    const usersProcessed = buckets.map((bucket) => {
+      const topHit: PrivTopHit | undefined = bucket.latest_doc_for_user?.hits?.hits?.[0];
+      const isPriv =
+        topHit?.fields?.['user.is_privileged']?.[0] ??
+        topHit?._source?.user?.is_privileged ??
+        false;
+      return {
+        username: bucket.key.username,
+        existingUserId: existingUserMap.get(bucket.key.username),
+        isPrivileged: Boolean(isPriv),
+        latestDocForUser: topHit,
+        labels: generateLabels(matchers, topHit),
+      };
+    });
+    return usersProcessed;
+  };
+
+  const generateLabels = (matchers: Matcher, topHit: PrivTopHit): Record<string, unknown> => {
+    // Implement label generation logic based on matchers and topHit (latest document for user)
```

Comment:
maybe add TODO: in front of this and any other stubbed out code comment so we can grep for it later, not a big deal though